src/app.js

```diff
@@ -3,40 +3,73 @@ import Form from './form.js'
 import Nav from './nav-bar.js'
 import CardList from './card.js'
 import {NoCards} from './card.js'
+import EditCard from './edit.js'
 
 export default class App extends React.Component {
   constructor(props) {
     super(props)
     const cardJSON = window.localStorage.getItem('cards')
+    const editJSON = window.localStorage.getItem('edit')
     this.state = {
       cards: JSON.parse(cardJSON) || [],
-      path: window.location.hash.replace(/#/g, '')
+      path: window.location.hash.replace(/#/g, ''),
+      edit: JSON.parse(editJSON) || []
     }
     this.handleSubmit = this.handleSubmit.bind(this)
+    this.handleSubmitEdit = this.handleSubmitEdit.bind(this)
+    this.handleClick = this.handleClick.bind(this)
   }
 
   renderView() {
     switch (this.state.path) {
       case 'create-card':
         return <Form handleSubmit={this.handleSubmit} />
+      case 'edit-card':
+        return <EditCard edit={this.state.edit} handleSubmitEdit={this.handleSubmitEdit} card={this.state.cards} />
     }
   }
 
   handleSubmit(event) {
     event.preventDefault()
+    const cards = this.state.cards.slice()
     const card = {}
     const createForm = new FormData(event.target)
 
-    if (createForm.get('question') === '' || createForm.get('answer') === '') {
-      return null
-    }
+    if (createForm.get('question') === '' || createForm.get('answer') === '') return null
     else {
+      card.id = this.state.cards.length + 1
       card.question = createForm.get('question')
       card.answer = createForm.get('answer')
-      this.state.cards.push(card)
-      event.target.reset()
+      cards.push(card)
+      window.location.hash = 'card-list'
+      this.setState({
+        cards: cards
+      })
+    }
+    event.target.reset()
+  }
+
+  handleClick(event) {
+    this.state.cards.map(card => {
```

Comment:
This is not what `map` is for.
> The map() method creates a new array with the results of calling a provided function on every element in the calling array.

Try [`Array.prototype.find`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find) instead.
> The find() method returns the value of the first element in the array that satisfies the provided testing function. Otherwise undefined is returned.